content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,487 @@
+        div
+            [ class "container" ]
+            [ ul
+                [ class "links" ]
+                [ li
+                    []
+                    [ a
+                        [ href "http://www.servicedenuages.fr/" ]
+                        [ text "Blog" ]
+                    ]
+                , li
+                    []
+                    [ a
+                        [ href "http://www.elm-lang.org/" ]
+                        [ text "Elm lang" ]
+                    ]
+                ]
+            ]
+
+
+Adding some state
+=================
+
+Now that you know how to render your page in HTML, it is time to start
+a program that knows how to handle events.
+
+The way Elm handles that is by having:
+
+- a ``Model``, an Elm record: a bit like a JavaScript object with
+  properties, that keep the state of the app,
+- an update function that will handle all the app events
+- a view function that will return the Virtual DOM that matches the state.
+
+The events and their params are defined in a ``Msg`` type. It is a bit
+like a enum that would take parameters.
+
+In order to do that we can use the ``beginnerProgram`` from the Html package:
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html exposing (..)
+
+
+    type Msg
+        = Noop
+
+
+    type alias Model =
+        { name : String }
+
+
+    main =
+        beginnerProgram { model = { name = "Rémy" }, view = view, update = update }
+
+
+    update : Msg -> Model -> Model
+    update msg model =
+        model
+
+
+    view : Model -> Html Msg
+    view model =
+        text ("Hello " ++ model.name)
+
+We can now handle an event and change the name when we click on it.
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html exposing (..)
+    import Html.Attributes exposing (..)
+    import Html.Events exposing (..)
+
+
+    type Msg
+        = Switch
+
+
+    type alias Model =
+        { name : String }
+
+
+    main =
+        beginnerProgram { model = { name = "Rémy" }, view = view, update = update }
+
+
+    update : Msg -> Model -> Model
+    update msg model =
+        case msg of
+            Switch ->
+                { model | name = "Séverine" }
+
+
+    view : Model -> Html Msg
+    view model =
+        div []
+            [ text "Hello "
+            , a [ href "#", onClick Switch ] [ text model.name ]
+            ]
+
+You can refresh the page and try it.
+
+If we want to switch back to ``Rémy`` when we click on ``Séverine`` we can add a ``if``:
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html exposing (..)
+    import Html.Attributes exposing (..)
+    import Html.Events exposing (..)
+
+
+    type Msg
+        = Switch
+
+
+    type alias Model =
+        { name : String }
+
+
+    main =
+        beginnerProgram { model = { name = "Rémy" }, view = view, update = update }
+
+
+    update : Msg -> Model -> Model
+    update msg model =
+        case msg of
+            Switch ->
```

Comment:
I'd personally immediately add a type parameter like `name`, especially as you mentioned this capability early in the post.